specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-09-01-preview/Overview.json

```diff
@@ -0,0 +1,397 @@
+            "type": "string"
+          },
+          "type": "object"
+        },
+        "closedIncidentsClassification": {
+          "description": "Aggregation of all closed incidents by closed classification reason: 'TruePositive', 'FalsePositive', 'BenignPositive', 'Undetermined', ...",
+          "additionalProperties": {
+            "description": "The amount of closed incidents with a given closed classification reason",
+            "type": "string"
+          },
+          "type": "object"
+        },
+        "incidentsStatusByCreationTime": {
+          "description": "A series of incidents by status",
+          "items": {
+            "$ref": "#/definitions/IncidentsByStatus",
+            "description": "Aggregation of all incidents by status in a specific time"
+          },
+          "type": "array",
+          "x-ms-identifiers": []
+        },
+        "meanTimeToAck": {
+          "description": "The mean time it took to acknowledge an incident",
+          "$ref": "#/definitions/MeanTimeToAction"
+        },
+        "meanTimeToClose": {
+          "description": "The mean time it took to close an incident",
+          "$ref": "#/definitions/MeanTimeToAction"
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "Incidents"
+    },
+    "IngestedEventCount": {
+      "type": "string"
+    },
+    "DataConnectorsSectionData": {
+      "description": "Data connectors overview section data",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SectionData"
+        }
+      ],
+      "properties": {
+        "activeConnectors": {
+          "description": "A dictionary of all active connectors",
+          "additionalProperties": {
+            "description": "Connector name",
+            "type": "string"
+          },
+          "type": "object"
+        },
+        "anomaliesVolume": {
+          "description": "A series of anomaly ingested events",
+          "items": {
+            "$ref": "#/definitions/IngestedEventCount",
+            "description": "The amount of record and their ingested time as string"
+          },
+          "type": "array"
+        },
+        "tiByType": {
+          "description": "Aggregation of all threat intelligence ingested events by type: 'URL', 'IP', 'Email', 'Other', ...",
+          "additionalProperties": {
+            "description": "The amount of events of a given type",
+            "type": "string"
+          },
+          "type": "object"
+        },
+        "totalVolume": {
+          "description": "A series of all ingested events",
+          "items": {
+            "$ref": "#/definitions/IngestedEventCount",
+            "description": "The amount of record and their ingested time as string"
+          },
+          "type": "array"
+        },
+        "unhealthyConnectors": {
+          "description": "A dictionary of all unhealthy connectors",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

use of additionalproperties needs to be avoided wherever an alternate typed schema is possible. Can u provide a reasoning for each of the usages of additionalproperties here on why it cannot be converted into a specific type?

Reply:
Yes,
in all the cases here, we are aggregating users data by a variety of parameters (status, healthiness, severity, . because we don't have any control\knowledge regarding the different parameters value nor the user data itself, we calculate a dictionary as response.